src/components/form/stepper/index.tsx

```diff
@@ -0,0 +1,144 @@
+import cx from 'classnames';
+import { Icon, Text } from 'components';
+import * as React from 'react';
+import { useMode, useTheme } from 'utils/hooks';
+
+import { InputComponent, InputPropsType, WrapperProps } from '../../../utils/types';
+import { StyledStepperInput } from './style';
+
+export interface StepperInputProps extends InputComponent, React.InputHTMLAttributes<HTMLInputElement> {
+  value: number;
+  onChange: any;
+  minValue: number;
+  maxValue: number;
+  step: number;
+  iconBefore?: HTMLObjectElement | JSX.Element;
+  iconAfter?: HTMLObjectElement | JSX.Element;
+  inputRef?: React.Ref<any>;
+  wrapperProps?: WrapperProps;
+  inputProps?: InputPropsType;
+}
+
+export const StepperInput = React.forwardRef<HTMLDivElement, StepperInputProps>(
+  (
+    {
+      value,
+      onChange,
+      minValue,
+      maxValue,
+      className,
+      title,
+      subTitle,
+      step,
+      disabled,
+      error,
+      wrapperProps,
+      inputProps,
+      inputRef,
+      ...props
+    }: StepperInputProps,
+    ref,
+  ): JSX.Element => {
+    const theme = useTheme();
+    const { mode } = useMode();
+
+    const [currentValue, setValue] = React.useState<number>(value);
+
+    const moreThenAllowed = currentValue > maxValue;
+    const lessThanAllowed = currentValue < minValue;
+
+    React.useEffect(() => {
+      if (onChange && !moreThenAllowed && !lessThanAllowed) {
+        onChange(currentValue);
+      }
+    }, [currentValue]);
+
+    const increment = () => {
+      setValue(Number(currentValue) + step);
+    };
+
+    const decrement = () => {
+      setValue(Number(currentValue) - step);
+    };
+
+    const handleChange = (event) => {
+      setValue(event.target.value);
```

Comment:
That's the right place to cast, as well catch NaN and also filter out non number inputs